src/data/metadata/MetadataD2ApiRepository.ts

```diff
@@ -101,11 +100,22 @@ export class MetadataD2ApiRepository implements MetadataRepository {
             if (dataStoreIds.length > 0) {
                 metadataPackage.dataStores = await d2ApiDataStore.getDataStore({ namespaces: ids });
             }
+            const dataStoresMetadata = await this.getDataStoresMetadata(ids);
+            const responseWithDataStores = { ...metadataPackage, ...dataStoresMetadata } as T;
 
-            return metadataPackage as T;
+            return responseWithDataStores;
         }
     }
 
+    private async getDataStoresMetadata(ids: Id[]) {
+        const d2ApiDataStore = new D2ApiDataStore(this.instance);
+        const dataStoreIds = DataStoreMetadata.getDataStoreIds(ids);
+        if (dataStoreIds.length === 0) return {};
+
+        const result = await d2ApiDataStore.getDataStore({ namespaces: dataStoreIds });
```

Comment:
`result` -> `dataStores`. the method name is a bit confusing, shouldn't it be plural?